relay-server/src/actors/spooler/mod.rs

```diff
@@ -415,9 +416,26 @@ impl OnDisk {
         let received_at: i64 = row
             .try_get("received_at")
             .map_err(BufferError::FetchFailed)?;
-        let start_time = StartTime::from_timestamp_millis(received_at as u64);
+        let start_time_instant = StartTime::from_timestamp_millis(received_at as u64).into_inner();
 
-        envelope.set_start_time(start_time.into_inner());
+        // This metric should show how long we waited till the envelope was unspooled.
+        relay_statsd::metric!(
+            timer(RelayTimers::EnvelopeOnDiskBufferTime) = start_time_instant.elapsed()
+        );
+
+        // If envelope stays longer than an 2 hours in on-disk spool, produce an error, so we could
+        // debug what kind of envelope is that.
+        if start_time_instant.elapsed() > Duration::from_secs(7200) {
```

Comment:
nit: this should allow removing the comment?
```suggestion
        if start_time_instant.elapsed() > Duration::from_secs(3600 * 2) {
```

Reply:
I like the comment for the reasoning, less for that 7200 = 2 hours